src/main/java/com/example/scholar_gql_poc/resolver/ScholarMutationResolver.java

```diff
@@ -25,4 +25,16 @@ public Scholar createScholar(String id,String name, int age){
             scholarRepository.save(sc);
             return scholarRepository.findById(id).get();
     }
+    public Scholar updateScholar(String id,String name, int age)
+    {
+        Optional<Scholar> scholar = scholarRepository.findById(id);
+        if (scholar.isPresent()) {
+            Scholar sc = scholar.get();
+            sc.setId(id);
+            sc.setName(name);
+            sc.setAge(age);
+            scholarRepository.save(sc);
+            return scholarRepository.findById(id).get();
+        }
+    }
```

Comment:
Can you please put piece implemented code in try and catch block and handle the exception